### PR TITLE
Correctly merge the root IDs with the child record IDs

### PR DESCRIPTION
**Pull request by leofeyer**
See https://github.com/contao/contao/pull/2036#pullrequestreview-461740840

### CODE REVIEW

**Comment by leofeyer** [comment]
This still does not work:

```
An exception occurred while executing 'SELECT id, pid FROM tl_calendar WHERE pid IN(7,2,4,6,13,11,10)':

SQLSTATE[42S22]: Column not found: 1054 Unknown column 'pid' in 'field list'
```

**Comment by leofeyer** [comment]
Ok, now it works in all cases (see d2a8183de0dcc48af9caf8ce8bfa39cbc1a0c3fc). @contao/reviewers Please review.

File: core-bundle/src/Resources/contao/drivers/DC_Table.php [hunk 1]
```diff
@@ -5609,7 +5609,22 @@ protected function filterMenu($intFilterPanel)
 			// Limit the options if there are root records
 			if (isset($GLOBALS['TL_DCA'][$table]['list']['sorting']['root']) && $GLOBALS['TL_DCA'][$table]['list']['sorting']['root'] !== false)
 			{
-				$arrProcedure[] = "id IN(" . implode(',', $this->Database->getChildRecords($GLOBALS['TL_DCA'][$table]['list']['sorting']['root'], $this->strTable)) . ")";
+				$rootIds = array_map('\intval', $GLOBALS['TL_DCA'][$table]['list']['sorting']['root']);
+
+				// Also add the child records of the table (see #1811)
+				if ($GLOBALS['TL_DCA'][$table]['list']['sorting']['mode'] == 5)
```
Comment (aschempp) [inline core-bundle/src/Resources/contao/drivers/DC_Table.php:5615]: This should be done in mode 6 as well.

**Review by aschempp — CHANGES_REQUESTED** [review]

File: core-bundle/src/Resources/contao/drivers/DC_Table.php (hunk 1 above)
Comment (leofeyer) [reply core-bundle/src/Resources/contao/drivers/DC_Table.php:5615]: No, this is fine the way it is. By using `$table` instead of `$this->strTable`, we are already accounting for mode 6.

Comment (aschempp) [reply core-bundle/src/Resources/contao/drivers/DC_Table.php:5615]: what if the parent table of the current is mode 6? Not sure if that would even work though 😆

Comment (leofeyer) [reply core-bundle/src/Resources/contao/drivers/DC_Table.php:5615]: Can you provide a test case to reproduce? Otherwise we should probably merge the PR and see if there are any edge cases, because this is blocking the release of Contao 4.4.51.

**Review by aschempp — APPROVED** [review]
